samples/file-encryptor/enclave/encryptor.cpp

```diff
@@ -64,7 +47,7 @@ int ecall_dispatcher::initialize(
         goto exit;
     }
     // init iv
-    memcpy(m_operating_iv, m_original_iv, IV_SIZE);
+    memcpy(m_operating_iv, header->salt, IV_SIZE);
```

Comment:
Also check the header is null or not here.
Is there any case that we can't use `m_header`? 
This may be a bit off-topic, can be addressed in a separate PR:
There seems a bit inconsistency between `initialize` and `process_encryption_header`. For example, both of them set `m_encrypt`. Can we have the `initialize` function to initialize `m_passord`, `m_encrypt`, and `m_header` instead?

Reply:
Added null check and moved m_password and m_header initialization to initialize().

Reply:
Should we use `m_header` instead of `header` here?

Reply:
replaced header with m_header.